com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceShadows.cs

```diff
@@ -0,0 +1,206 @@
+            internal Material material;
+
+            // Private Variables
+            private ScreenSpaceShadowsSettings m_CurrentSettings;
+            private RenderTextureDescriptor m_RenderTextureDescriptor;
+            private RenderTargetHandle m_RenderTarget;
+
+            // Constants
+            private const string k_SSShadowsTextureName = "_ScreenSpaceShadowmapTexture";
+
+            internal ScreenSpaceShadowsPass()
+            {
+                m_CurrentSettings = new ScreenSpaceShadowsSettings();
+                m_RenderTarget.Init(k_SSShadowsTextureName);
+            }
+
+            internal bool Setup(ScreenSpaceShadowsSettings featureSettings)
+            {
+                m_CurrentSettings = featureSettings;
+                ConfigureInput(ScriptableRenderPassInput.Depth);
+
+                return material != null;
+            }
+
+            /// <inheritdoc/>
+            public override void OnCameraSetup(CommandBuffer cmd, ref RenderingData renderingData)
+            {
+                m_RenderTextureDescriptor = renderingData.cameraData.cameraTargetDescriptor;
+                m_RenderTextureDescriptor.depthBufferBits = 0;
+                m_RenderTextureDescriptor.msaaSamples = 1;
+                m_RenderTextureDescriptor.graphicsFormat = RenderingUtils.SupportsGraphicsFormat(GraphicsFormat.R8_UNorm, FormatUsage.Linear | FormatUsage.Render)
+                    ? GraphicsFormat.R8_UNorm
+                    : GraphicsFormat.B8G8R8A8_UNorm;
+
+                cmd.GetTemporaryRT(m_RenderTarget.id, m_RenderTextureDescriptor, FilterMode.Point);
+
+                RenderTargetIdentifier renderTargetTexture = m_RenderTarget.Identifier();
+                ConfigureTarget(renderTargetTexture);
+                ConfigureClear(ClearFlag.All, Color.white);
+            }
+
+            /// <inheritdoc/>
+            public override void Execute(ScriptableRenderContext context, ref RenderingData renderingData)
+            {
+                if (material == null)
+                {
+                    Debug.LogErrorFormat("{0}.Execute(): Missing material. {1} render pass will not execute. Check for missing reference in the renderer resources.", GetType().Name, profilerTag);
+                    return;
+                }
+
+                Camera camera = renderingData.cameraData.camera;
+                ShadowData shadowData = renderingData.shadowData;
+
+                int cascadesCount = shadowData.mainLightShadowCascadesCount;
+
+                CommandBuffer cmd = CommandBufferPool.Get();
+                using (new ProfilingScope(cmd, ProfilingSampler.Get(URPProfileId.ResolveShadows)))
+                {
+                    cmd.SetViewProjectionMatrices(Matrix4x4.identity, Matrix4x4.identity);
+                    cmd.DrawMesh(RenderingUtils.fullscreenMesh, Matrix4x4.identity, material);
+                    cmd.SetViewProjectionMatrices(camera.worldToCameraMatrix, camera.projectionMatrix);
```

Comment:
This might cause some issues with XR. @thomas-zeng 
 We do have internal functions to set camera matrices:
 `RenderingUtils.SetViewAndProjectionMatrices(cmd, cameraData.GetViewMatrix(), cameraData.GetGPUProjectionMatrix(), false);`

Reply:
Ok, thanks. let me use it